### PR TITLE
Record stringified property as `""` when `ToString()` returns `null` in JsonFormatters

### PR DESCRIPTION
**Pull request by rafaelsc**
**What issue does this PR address?**
This will fix the issue #1451 - A bad designed Class can cause an Exception when using Stringify with JsonFormatter or JsonValueFormatter.

**Does this PR introduce a breaking change?**
No

**Please check if the PR fulfills these requirements**
- [ x ] The commit follows our [guidelines](https://github.com/serilog/serilog/blob/dev/CONTRIBUTING.md)
- [ x ] Unit Tests for the changes have been added (for bug fixes / features)

### CODE REVIEW

**Comment by rafaelsc** [comment]
Related Issue #1426
Related PF #1427

**Comment by rafaelsc** [comment]
Bonus: More unit testing

File: src/Serilog/Capturing/PropertyValueConverter.cs [hunk 1]
```diff
@@ -299,7 +299,7 @@ bool TryConvertCompilerGeneratedType(object value, Destructuring destructuring,
         LogEventPropertyValue Stringify(object value)
         {
             var stringified = value.ToString();
-            var truncated = stringified == null ? null : TruncateIfNecessary(stringified);
+            var truncated = stringified is null ? null : TruncateIfNecessary(stringified);
```
Comment (rafaelsc) [inline src/Serilog/Capturing/PropertyValueConverter.cs:302]: Performance Improvements in property destruction

File: test/Serilog.Tests/Formatting/Json/JsonFormatterTests.cs [hunk 2]
```diff
@@ -225,5 +225,51 @@ public void SequencesOfSequencesAreSerialized()
...
             Assert.Equal("Hello", h);
         }
+
+
+        [Fact]
+        public void ABadBehaviorClassAsSerialized()
+        {
+            var p = new MessageTemplateParser();
+            var e = new LogEvent(Some.OffsetInstant(), LogEventLevel.Information, null,
+                p.Parse("{@ABadBehavior}"), new[] { new LogEventProperty("ABadBehavior", new ScalarValue(new ABadBehavior())) });
```
Comment (rafaelsc) [inline test/Serilog.Tests/Formatting/Json/JsonFormatterTests.cs:235]: Now I think I created this UnitTes wrongly. The ScalarValue can never have an instance of an object like this in the normal execution of the Serilog pipeline.

File: test/Serilog.Tests/Formatting/Json/JsonFormatterTests.cs [hunk 3]
```diff
@@ -225,5 +225,51 @@ public void SequencesOfSequencesAreSerialized()
...
+            var h = (string)d.Properties.ABadBehavior;
+            Assert.Null(h);
+        }
+
+        [Fact]
+        public void ABadBehaviorClassAsStringification()
+        {
+            var p = new MessageTemplateParser();
+            var e = new LogEvent(Some.OffsetInstant(), LogEventLevel.Information, null,
+                p.Parse("{$ABadBehavior}"), new[] { new LogEventProperty("ABadBehavior", new ScalarValue(new ABadBehavior())) });
```
Comment (rafaelsc) [inline test/Serilog.Tests/Formatting/Json/JsonFormatterTests.cs:248]: Now I think I created this UnitTes wrongly. The ScalarValue can never have an instance of an object like this in the normal execution of the Serilog pipeline.

File: test/Serilog.Tests/Formatting/Json/JsonFormatterTests.cs [hunk 4]
```diff
@@ -225,5 +225,51 @@ public void SequencesOfSequencesAreSerialized()
...
+            var h = (string)d.Properties.ABadBehavior;
+            Assert.Null(h);
+        }
+
+        [Fact]
+        public void ABadBehaviorClassAsRenderings()
+        {
+            var p = new MessageTemplateParser();
+            var e = new LogEvent(Some.OffsetInstant(), LogEventLevel.Information, null,
+                p.Parse("{ABadBehavior}"), new[] { new LogEventProperty("ABadBehavior", new ScalarValue(new ABadBehavior())) });
```
Comment (rafaelsc) [inline test/Serilog.Tests/Formatting/Json/JsonFormatterTests.cs:261]: Now I think I created this UnitTes wrongly. The ScalarValue can never have an instance of an object like this in the normal execution of the Serilog pipeline.

**Comment by rafaelsc** [comment]
Update to have the same behavior of #1467

**Comment by nblumhardt** [comment]
That's great 👍 I'm going to try to get the wheels turning again and get some things merged :-) -- Is this ready to go?

**Comment by rafaelsc** [comment]
Can you wait a couple of days? Now that I have a better knowledge about Serilog Core I think some tests that I created are wrong. And I will try to fix them asap.

**Comment by rafaelsc** [comment]
@nblumhardt I choose to remove some tests that I added because the way that I created the test was impossible to do in normal Serilog operation.

**Comment by nblumhardt** [comment]
(Misclicked - meant to squash and merge! Will do in future :-) )

**Pull request merged**